docs/source/_templates/txn_submit_tutorial.rst

```diff
@@ -0,0 +1,484 @@
+tampered with, a hash of it must be included within the Transaction's header.
+This hash should be created using the SHA-512 function, and then formatted as a
+hexadecimal string.
+
+{% if language == 'JavaScript' %}
+
+.. code-block:: javascript
+
+    const sha512 = require('crypto').createHash('sha512')
+
+    const payloadSha512 = sha512.update(payloadBytes).digest('hex')
+
+{% else %}
+
+.. code-block:: python
+
+    from hashlib import sha512
+
+    payload_sha512 = sha512(payload_bytes).hexdigest()
+
+{% endif %}
+
+
+2. Build the TransactionHeader
+------------------------------
+
+Transactions and their headers are built using
+`Google Protocol Buffer <https://developers.google.com/protocol-buffers/>`_
+(or Protobuf) format. This allows data to be serialized and deserialzed
+consistently and efficiently across multiple platforms and multiple languages.
+The Protobuf definition files are located in the
+`/protos <https://github.com/hyperledger/sawtooth-core/tree/master/protos>`_,
+directory at the root level of the sawtooth-core repo. These files will have to
+first be compiled into usable {{ language }} classes. Then, serializing a
+*TransactionHeader* is just a matter of plugging the right data into the right
+keys.
+
+{% if language == 'JavaScript' %}
+
+.. code-block:: javascript
+
+    const protobuf = require('protobufjs')
+
+    const txnRoot = protobuf.loadSync('protos/transactions.proto')
+    const TransactionHeader = txnRoot.lookup('TransactionHeader')
+
+    const txnHeaderBytes = TransactionHeader.encode({
+        batcherPubkey: publicKeyHex,
+        dependencies: [],
+        familyName: 'intkey',
+        familyVersion: '1.0',
+        inputs: ['1cf126'],
+        nonce: Math.random().toString(36),
+        outputs: ['1cf126'],
+        payloadEncoding: 'application/cbor',
+        payloadSha512: payloadSha512,
+        signerPubkey: publicKeyHex
+    }).finish()
+
+{% else %}
+
+.. code-block:: bash
+
+    % protoc --python-out=compiled_protos/ protos/transactions.proto
+
+.. code-block:: python
+
+    from random import randint
+    from compiled_protos.transactions_pb2 import TransactionHeader
+
+    txn_header = TransactionHeader(
+        batcher_pubkey=public_key_hex,
+        dependencies=[],
+        family_name='intkey',
+        family_version='1.0',
+        inputs=['1cf126'],
+        nonce=str(randint(0, 1000000000)),
+        outputs=['1cf126'],
+        payload_encoding='application/cbor',
+        payload_sha512=payload_sha512,
+        signer_pubkey=public_key_hex)
+
+    txn_header_bytes = txn_header.SerializeToString()
+
+{% endif %}
+
+Remember that *inputs* and *outputs* are state addresses that the Transaction is
+allowed to read from or write to (see TransactionHeaders in
+:doc:`/architecture/transactions_and_batches`). In this case we used a six
+character prefix rather than full 70-character addresses, allowing us to work
```

Comment:
I wondered if it was. Mostly I just I didn't want to type out a 70-character address, and that wouldn't be a realistic way to use a TransactionEncoder anyway. A specific address would probably be set on a Transaction by Transaction basis. Probably best to make it more accurate. Will think on it.